haystack/nodes/retriever/dense.py

```diff
@@ -2254,7 +2207,7 @@ def retrieve_batch(
                     " as queries or a single filter that will be applied to each query."
                 )
         else:
-            filters = [filters] * len(queries) if filters is not None else [{}] * len(queries)
```

Comment:
I think we need this. Otherwise `for batch, cur_filters in zip(batches, filters):` will throw an error if filters is None (see comment above).